### PR TITLE
[CORE-632] Add workflow to check sample_pregenesis is uptodate; sort pregenesis file so ordering is deterministic

### PR DESCRIPTION
**Pull request by teddyding**
### Changelist
- Add workflow to check sample_pregenesis is uptodate
- Sort pregenesis file so ordering is deterministic

### Test Plan
Check workflow on this PR. 

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/scripts/genesis/prod_pregenesis.sh [hunk 1]
```diff
@@ -170,6 +170,11 @@ create_pregenesis_file() {
 	overwrite_genesis_production
 }
 
+sort_genesis_file(){
+	jq -S . $VAL_CONFIG_DIR/genesis.json > $VAL_CONFIG_DIR/sorted_genesis.json
```
Comment (teddyding) [inline protocol/scripts/genesis/prod_pregenesis.sh:174]: Sorted the output json so ordering is deterministic and can be diff'ed

File: .github/workflows/protocol-pregenesis.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+on:  # yamllint disable-line rule:truthy
+  pull_request:
+    paths:
+      - 'protocol/scripts/genesis/**'
+      - 'protocol/testing/**'
+  push:
+    branches:
+      - main
+      - 'release/protocol/v0.[0-9]+.x'  # e.g. release/protocol/v0.1.x
```
Comment (ttl33) [inline .github/workflows/protocol-pregenesis.yml:11]: for my understanding, are we going to use the letter `x`? I thought it was just a placeholder

Comment (teddyding) [reply .github/workflows/protocol-pregenesis.yml:11]: The branch name explicitly has letter `x`, like `release/protocol/v1.x`

File: protocol/scripts/genesis/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# NB: This is a digest for a multi-arch manifest list, you will want to get this by running
```
Comment (roy-dydx) [inline protocol/scripts/genesis/Dockerfile:1]: What's the difference between this dockerfile and `protocol/Dockerfile`? Can we avoid the duplication and extra maintenance by reusing it like we do in the staging dockerfile, localnet dockerfile, etc?

If the only difference is installing jq and dasel, it's OK to add it to the original docker image IMO.

Comment (teddyding) [reply protocol/scripts/genesis/Dockerfile]: Updated to use the `dydxprotoocl-base` image as base

**Pull request merged**